src/wp-includes/pluggable.php

```diff
@@ -2624,90 +2624,186 @@ function wp_hash( $data, $scheme = 'auth', $algo = 'md5' ) {
 	 * instead use the other package password hashing algorithm.
 	 *
 	 * @since 2.5.0
+	 * @since x.y.z The password is now hashed using bcrypt instead of phpass.
 	 *
-	 * @global PasswordHash $wp_hasher PHPass object.
+	 * @global PasswordHash $wp_hasher phpass object.
 	 *
 	 * @param string $password Plain text user password to hash.
 	 * @return string The hash string of the password.
 	 */
 	function wp_hash_password( $password ) {
 		global $wp_hasher;
 
-		if ( empty( $wp_hasher ) ) {
-			require_once ABSPATH . WPINC . '/class-phpass.php';
-			// By default, use the portable hash from phpass.
-			$wp_hasher = new PasswordHash( 8, true );
+		if ( ! empty( $wp_hasher ) ) {
+			return $wp_hasher->HashPassword( trim( $password ) );
+		}
+
+		if ( strlen( $password ) > 4096 ) {
```

Comment:
Footgun.

This is not part of the public API and many plugins use wp_hash_password for non-passwords, sadly.

A PHP notice or error_log would probably be better.

Reply:
This is part of the public API because it matches the current behaviour of the phpass implementation, which limits passwords to 4096 characters in length. I am not attached to it though, and the new use of sha384+base64+bcrypt shouldn't be subject to DoS concerns from overly long passwords. Shall we pull it out?

Refs:

* https://github.com/WordPress/wordpress-develop/blob/09459f1e4b5eeb3604cfedfa12047ec03c7489b5/src/wp-includes/class-phpass.php#L216-L218
* https://github.com/WordPress/wordpress-develop/blob/09459f1e4b5eeb3604cfedfa12047ec03c7489b5/src/wp-includes/class-phpass.php#L249-L251

Reply:
You're right, it's implicitly exposed as the public API if phpass is used